### PR TITLE
Job state stored in distributed cache in MR mode.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@chavdar @sahilTakiar can you review? This addresses https://github.com/linkedin/gobblin/issues/911

File: gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java [hunk 1]
```diff
@@ -313,8 +314,10 @@ private void prepareHadoopJob(List<WorkUnit> workUnits) throws IOException {
 
     // Serialize source state to a file which will be picked up by the mappers
     Path jobStateFilePath = new Path(this.mrJobDir, JOB_STATE_FILE_NAME);
-    SerializationUtils.serializeState(this.fs, jobStateFilePath, this.jobContext.getJobState(), 100);
+    SerializationUtils.serializeState(this.fs, jobStateFilePath, this.jobContext.getJobState());
+    DistributedCache.addCacheFile(jobStateFilePath.toUri(), this.job.getConfiguration());
```
Comment (sahilTakiar) [inline gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java]: We should use `MRJobLauncher.addHDFSFiles` which already does this, it also sets up a symlink to the file, this way the additional property `JOB_STATE_DISTRIBUTED_CACHE_NAME` should not longer be necessary.

Using the symlink should also simply the code added to the `setup` method.

**Comment by ibuenros** [comment]
@sahilTakiar I tried using `MRJobLauncher.addHDFSFiles` but I couldn't get it working. `DistributedCache.getLocalCacheFiles(context.getConfiguration())` throws an NPE, and trying to simply load `new File(name)` cannot find the file.
Any suggestions?

**Comment by ibuenros** [comment]
Talked with @sahilTakiar offline. Merging.

**Pull request merged**